src/node/desktop/src/main/gwt-callback.ts

```diff
@@ -902,10 +903,31 @@ export class GwtCallback extends EventEmitter {
     });
 
     ipcMain.handle('desktop_startup_error_info', async (event, varName: string) => {
+      if (varName == 'launch_failed') {
+        this.addMacOSVersionError();
+      }
       return resolveTemplateVar(varName, this.errorPageData);
     });
   }
 
+  addMacOSVersionError(): void {
+    if (platform() === 'darwin') {
+      const release_major = parseInt(release().substring(0, release().indexOf('.')));
+      // macOS 11.0 uses darwin 20.0.0
+      if (release_major < 20) {
+        let versionError = 'You are using an unsupported operating system.' +
+        ' RStudio requires macOS 11 or higher.';
```

Comment:
I wonder if we should say "macOS 11 (Big Sur)". A lot of Mac users use the names rather than the version numbers, I think, so including both would be helpful.